manic/repository_git.py

```diff
@@ -282,36 +303,40 @@ def _create_remote_name(self):
         remote_name = "{0}_{1}".format(base_name, repo_name)
         return remote_name
 
-    def _checkout_ref(self, repo_dir, verbosity):
+    def _checkout_ref(self, repo_dir, verbosity, submodules):
         """Checkout the user supplied reference
+        if <submodules> is True, recursively initialize and update
+        the repo's submodules
         """
         # import pdb; pdb.set_trace()
         cwd = os.getcwd()
         os.chdir(repo_dir)
         if self._url.strip() == LOCAL_PATH_INDICATOR:
-            self._checkout_local_ref(verbosity)
+            self._checkout_local_ref(verbosity, submodules)
         else:
-            self._checkout_external_ref(verbosity)
+            self._checkout_external_ref(verbosity, submodules)
+
         os.chdir(cwd)
 
-    def _checkout_local_ref(self, verbosity):
+    def _checkout_local_ref(self, verbosity, submodules):
         """Checkout the reference considering the local repo only. Do not
         fetch any additional remotes or specify the remote when
         checkout out the ref.
-
+        if <submodules> is True, recursively initialize and update
+        the repo's submodules
         """
         if self._tag:
             ref = self._tag
         elif self._branch:
             ref = self._branch
-        else:
```

Comment:
Can you explain the deletion of this `else` block? My initial thought is that maybe it needs to be changed to `elif self._hash`, but should still be kept here. I haven't considered this carefully, but I'm wary of its deletion.

Reply:
Wow, my first reaction is -- that cannot possibly work that way (`_check_for_valid_ref` has to fail, right?). But since tests pass, that means there is no test! I am not sure how to write a test for that (need checkout called with a url of `.`) so I just put the code back (I have no idea how it disappeared in the first place).
Thanks for the catch!